octoprint_octorelay/__init__.py

```diff
@@ -229,27 +233,33 @@ def on_api_command(self, command, data):
                     activeRelays.append(relaydata)
             return flask.jsonify(activeRelays)
 
-        index = data['pin']
+        # added api command to get led status
+        if command == "getStatus":
+            ledState = self.update_relay(data["pin"], get_status=True)
+            return flask.jsonify(status=ledState)
 
-        settings = self.get_settings_defaults()[index]
-        settings.update(self._settings.get([index]))
+        if command == "update":
+            status = self.update_relay(data["pin"])
+            return flask.jsonify(status=status)
 
-        relay_pin = int(settings["relay_pin"])
-        inverted = settings['inverted_output']
-        cmdON = settings['cmdON']
-        cmdOFF = settings['cmdOFF']
+    def update_relay(self, index, get_status=False):
```

Comment:
I'm leaning against the approach:
the function called `update_relay` being called with `get_status` parameter does NOT update the relay (lines 260-261)

Reply:
fixed in 1ab39eac9dabad030ef136d16df0124629763259